### ISSUE TITLE
Wrong FK's creation in migration

### ISSUE THREAD

**Issue by berets76** [aspnet/EntityFrameworkCore#11235]
Hi, we upgrade to 2.1.0-preview1-final to solve some issue related to "Optimization of correlated subqueries" (2.1.0 release note).

Our model was perfectly generated with 2.0.1 version but now the core seems to not recognize fluent settings and duplicate our foreign keys.

### Steps to reproduce

Sample model

```C#
    [Table("Countries")]
    public class Country
    {
        [Key,Column(Order = 0)]
        public Guid TenantID { get; set; }
        [Key, Column(Order = 1)]
        public Guid ID { get; set; }
        [StringLength(3)]
        public string NumericCode { get; set; }
        [StringLength(3)]
        public string IsoCode { get; set; }
        public List<CountryName> Names { get; set; }
    }

    [Table("CountryNames")]
    public class CountryName
    {
        [Key, Column(Order = 0)]
        public Guid TenantID { get; set; }
        [Key,Column(Order = 1)]
        public Guid CountryID { get; set; }
        [Key, Column(Order = 2)]
        [StringLength(3)]
        public string LanguageID { get; set; }
        [StringLength(255)]
        public string Name { get; set; }

        public Country Country { get; set; }
    }
```
Fluent settings

```C#
            // Country
            modelBuilder.Entity<Country>().HasKey(c => new { c.TenantID, c.ID });
            modelBuilder.Entity<Country>().Property(p => p.ID).ValueGeneratedOnAdd().HasDefaultValueSql("newsequentialid()");
            // CountryName
            modelBuilder.Entity<CountryName>().HasKey(c => new { c.TenantID, c.CountryID, c.LanguageID });
            modelBuilder.Entity<CountryName>().HasOne<Country>().WithMany(wm => wm.Names).HasForeignKey(fk => new { fk.TenantID, fk.CountryID });
            modelBuilder.Entity<CountryName>().HasQueryFilter(qf => qf.LanguageID == LanguageID);
```

Generated CountryNames table

![image](https://user-images.githubusercontent.com/9038856/37277223-639aa482-25e4-11e8-8efb-6ed18d49cf91.png)

With 2.01 version that table was generated correctly.

In addition to this, we have an huge list of "warning" (already reported in other issues) about GUID's incompatiblities (not correct because all ours ID are GUID) : don't clear if could be related.

> The best match for foreign key properties {'ID' : Guid} are incompatible with the principal key {'ID' : string}.
> Microsoft.EntityFrameworkCore.Model[10602]

### Further technical details
EF Core version: v2.1.0-preview1-final
Database Provider: Microsoft.EntityFrameworkCore.SqlServer
Operating system: Windows 10 Enterprise
IDE: Visual Studio Enterprise 2017 15.6.1 Preview 1

**Comment by berets76** [comment]
### UPDATE (SOLVED?)

The problem is in the fluent, this line 

`modelBuilder.Entity<CountryName>().HasOne<Country>().WithMany(wm => wm.Names).HasForeignKey(fk => new { fk.TenantID, fk.CountryID });`

generate "wrong" FK, instead this one

`modelBuilder.Entity<CountryName>().HasOne(ho => ho.Country).WithMany(wm => wm.Names).HasForeignKey(fk => new { fk.TenantID, fk.CountryID });`

use my correct fields.

The "bug" is that with EF Core 2.0.1 previous lines are equivalent and generate the same table, fields and keys.

Maybe it's an unexpected behavior in 2.0.1

**Comment by ajcvickers** [comment]
Notes from triage: The behavior, as designed, for the Has.../With... methods is that if no navigation property is specified, then it means that relationship explicitly does not use that navigation property. However, there was a bug in 2.0 and before which meant that if the navigation property was not explicitly configured to be part of another relationship, then it would be brought in as the navigation to use anyway.

At this point in time we are leaving the fixed code in place for 2.1, since it is consistent with how other APIs behave and we believe the best approach is for people to update their code if they hit this issue after updating to 2.1. However, we recognize that a log of people may hit this, and it would be good to know who does. We may even consider allowing the old behavior to be turned back on in some way, even though it is technically a bug.

Also filed #11270 to track making the API docs clearer here.
